proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -105,9 +105,9 @@ func callMethod(p: P) {
 }
 ```
 
-Evidently, an associated type or other unfortunate requirement (such as `bar`) cannot speak for the rest of the interface. Some protocols still have a useful subset of functionality that either doesn't rely on `Self` and associated types, or relies on them in an existentially compatible way.
+Evidently, an associated type or other unfortunate requirements (such as `bar`) cannot speak for the rest of the interface. Some protocols still have a useful subset of functionality that either doesn't rely on `Self` and associated types, or that relies on them in a way compatible with existential types.
```

Comment:
* "requirement" here relates to both "other" and  "associated type", since we want to distinguish an associated type requirement from an associated type. So the singular form was deliberate. We should either use plural on "associated type" as well, or keep both in singular form.
* If you want to use start both clauses with "that", you should place the currently shared "that" *after* "either" (hopefully makes sense).

```suggestion
Evidently, an associated type or other unfortunate requirement (such as `bar`) cannot speak for the rest of the interface. Some protocols still have a useful subset of functionality that either doesn't rely on `Self` and associated types, or relies on them in a way compatible with existential types.
```

Reply:
Oh, I see; find a bit confusing, though. How about:

Also some nits on the text. I'm not sure if I wrote some of these or if you did, but reading the text again I think some parts need to be more specific. 

1. Why are requirements such as `bar` characterized as unfortunate. Should we elaborate on this?
2. How –– specifically –– can a protocol be "compatible with existential values/types"?

----
Evidently, an associated type requirement or another unfortunate requirement (such as `bar`) cannot speak for the rest of the interface. Some protocols still have a useful subset of functionality that does not rely on `Self` and associated types, or is compatible with existential values.

Reply:
> 1. Why are requirements such as bar characterized as unfortunate. Should we elaborate on this?

This is the subject of the previous paragraph.

> 2. How –– specifically –– can a protocol be "compatible with existential values/types"?

Not a protocol though, its functionality. "Compatible" is meant to emphasize that a requirement may depend on `Self` in a way that does not contradict the criteria (and could have been used on an existential value were the protocol not under the restriction).